### PR TITLE
refactor: github and gitlab in pkg/util

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
- use the interface to Compatible with `gitlab` and `github`
- `github` repo support one commit in one pr
-  both `gitlab` and `github` use `git.RepoInfi` to init
-  extract `commitinfo` for more general usage
- delete hard-code code

## Related Issues
#987 

## New Behavior (screenshots if needed)
- Before `github` commit when using reposcaffolding
image

- Now `github` commit when using reposcaffolding
image
- tests
image

### CODE REVIEW

File: internal/pkg/plugininstaller/common/repo.go [hunk 1]
```diff
@@ -30,54 +31,36 @@ type Repo struct {
...
-}
-
-// buildgitlabOpts build gitlab connection options
-func (d *Repo) buildgitlabOpts() *gitlab.CreateProjectOptions {
-	return &gitlab.CreateProjectOptions{
-		Name:       d.Repo,
-		Branch:     d.getBranch(),
-		Namespace:  d.Org,
-		Visibility: d.Visibility,
+func (d *Repo) NewClient() (repo.ClientOperation, error) {
```
Comment (iyear) [inline internal/pkg/plugininstaller/common/repo.go:46]: The original code get client supports GitHub and GitLab, why does it only support GitHub here?

File: pkg/util/github/commit.go [hunk 2]
```diff
@@ -28,3 +24,17 @@ func (c *Client) GetLastCommit() (*github.RepositoryCommit, error) {
 
 	return commits[0], nil
 }
+
+func (c *Client) GetCommitTree(ref *github.Reference, commitInfo *repo.CommitInfo) (*github.Tree, error) {
```
Comment (iyear) [inline pkg/util/github/commit.go:28]: The function name and the action of the code do not match. `Get` <-> `CreateTree`. 

`BuildCommitTree` or `CreateCommitTree`?

File: pkg/util/github/github.go [hunk 3]
```diff
@@ -9,35 +9,30 @@ import (
...
 
 	"github.com/devstream-io/devstream/pkg/util/log"
+	"github.com/devstream-io/devstream/pkg/util/repo"
 )
 
 const (
-	DefaultWorkPath = ".github-workpath"
+	defaultWorkPath          = ".github-workpath"
+	defaultCommitAuthor      = "devstream"
+	defaultCommitAuthorEmail = "devstream@io"
```
Comment (iyear) [inline pkg/util/github/github.go:18]: Not a valid email address

File: pkg/util/types/bool.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
 package types
 
 func Bool(v bool) *bool { return &v }
+
+func String(v string) *string { return &v }
```
Comment (iyear) [inline pkg/util/types/bool.go:5]: Could change file name.

**Review by iyear — APPROVED** [review]
There are so many modified files that another person needs to review them and should be merged in after testing.

File: pkg/util/repo/commit.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+// GitFileInfo contains file local path and remote git path
+type GitFilePathInfo struct {
+	SourcePath       string
+	DestionationPath string
+}
+
+func GetFileContent(files []*GitFilePathInfo) GitFileContentMap {
+	gitFileMap := make(map[string][]byte)
+	for _, f := range files {
+		content, err := ioutil.ReadFile(f.SourcePath)
```
Comment (aFlyBird0) [inline pkg/util/repo/commit.go:29]: Please do not use `ioutil`. You could use `os.ReadFile` instead`.

**Review by aFlyBird0 — APPROVED** [review]

**Pull request merged**